sdf/bezier.go

```diff
@@ -24,79 +22,76 @@ type BezierPolynomial struct {
 }
 
 // Return the bezier polynomial function value.
-func (p *BezierPolynomial) f0(t float64) float64 {
+func (p *BezierPolynomial) f0(t float64) (float64, error) {
 	switch p.n {
 	case 0:
 		// point
-		return p.a
+		return p.a, nil
 	case 1:
 		// linear
-		return p.a + t*p.b
+		return p.a + t*p.b, nil
 	case 2:
 		// quadratic
-		return p.a + t*(p.b+t*p.c)
+		return p.a + t*(p.b+t*p.c), nil
 	case 3:
 		// cubic
-		return p.a + t*(p.b+t*(p.c+t*p.d))
+		return p.a + t*(p.b+t*(p.c+t*p.d)), nil
 	case 4:
 		// quartic
-		return p.a + t*(p.b+t*(p.c+t*(p.d+t*p.e)))
+		return p.a + t*(p.b+t*(p.c+t*(p.d+t*p.e))), nil
 	default:
-		log.Panicf("bad polynomial order %d", p.n)
-		return 0
+		return 0, fmt.Errorf("bad polynomial order %d", p.n)
```

Comment:
In this particular instance I think the panic is legitimately a panic.
Also: turning it into an error and propagating it back up the call chain is pretty ugly.
Also: I added it because I wanted the switch to have all cases- but p.n is set in the "set" function so actually panic-ing here as a result of the library user seems pretty small.

Reply:
looked like an oversight because 46 is dead code

Reply:
The compiler doesn't recognise the panic as terminating so you need a return to keep it happy, but yes it never gets called.

Reply:
reverted these in de810e8